### PR TITLE
Don't allow passing `load_in_8bit` and `load_in_4bit` at the same time

### PR DESCRIPTION
**Pull request by osanseviero**
# What does this PR do?

This PR disallows having both `load_in_8bit` and `load_in_4bit` set simultaneously. This would help avoid unexpected behaviors or broken configurations.

## Example

**Without the PR**

```
from transformers import BitsAndBytesConfig

BitsAndBytesConfig(load_in_8bit=True, load_in_4bit=True)
ValueError: load_in_4bit and load_in_8bit are both True, but only one can be used at the same time
```

works fine

**With the PR**

```
from transformers import BitsAndBytesConfig

BitsAndBytesConfig(load_in_8bit=True, load_in_4bit=True)
ValueError: load_in_4bit and load_in_8bit are both True, but only one can be used at the same time
```

errors as I would expect

### CODE REVIEW

**Review by younesbelkada — COMMENTED** [review]
Makes sense to me, indeed conceptually we can't load in 8bit and 4bit at the same time! 
cc @SunMarc @Titus-von-Koeller as well just FYI

File: tests/quantization/bnb/test_4bit.py
Comment (younesbelkada) [inline tests/quantization/bnb/test_4bit.py:655]:
```suggestion
```

**Review by younesbelkada — APPROVED** [review]
LGTM thanks for this!

File: tests/quantization/bnb/test_4bit.py
Comment (osanseviero) [reply tests/quantization/bnb/test_4bit.py:655]: Thanks! I had it for consistency with other tests - removing now!

**Review by SunMarc — APPROVED** [review]
LGTM !

**Comment by osanseviero** [comment]
Hey folks, should we merge this PR? :D

**Comment by younesbelkada** [comment]
Definitely, @Titus-von-Koeller if you don't see any issue what do you think of merging this PR?
cc @amyeroberts @ArthurZucker for a core-maintainer review before merging

**Review by ArthurZucker — APPROVED** [review]
Thanks!

**Pull request merged**
